pkg/storage/model_planStore.go

```diff
@@ -361,3 +366,88 @@ func (s *Store) ModelPlanDeleteByID(logger *zap.Logger, id uuid.UUID) (sql.Resul
 
 	return sqlResult, nil
 }
+
+func (s *Store) ModelPlanGetByOperationalSolutionKey(
+	logger *zap.Logger,
+	opSolKey models.OperationalSolutionKey,
+) ([]*model.ModelPlanAndOperationalSolution, error) {
+
+	stmt, err := s.db.PrepareNamed(modelPlanGetByOperationalSolutionKeySQL)
+	if err != nil {
+		return nil, err
+	}
+	defer stmt.Close()
+
+	arg := map[string]interface{}{
+		"operational_solution_key": opSolKey,
+	}
+
+	rows, err := stmt.Queryx(arg)
```

Comment:
Usage of Queryx and row-based iteration here is a matter of preference. I probably could have deserialized directly into a slice then decomposed the slice. Both ways work. Theoretically there could be some micro-considerations on mem stack performance, but realistically the number of models and opSols we'll see is very unlikely to be cause for concern here.

Reply:
> deserialized directly into a slice then decomposed the slice

Can you elaborate on this? I feel like with the way the `ModelPlanAndOperationalSolution` struct is designed we could just use a `sqlx.Select()` method to populate the slice of structs automatically (without any need for iterating over the rows, or the `models.New*Full()` methods)

Reply:
The constructors are a workaround for package scope issues I was running into with inline construction.

Connected to discussion on deserialization, unless we can find a way to scan embedded structs in a way which disambiguates fields we will still need to use *Raw as an intermediary struct to capture query results regardless of sqlx method used. Using Queryx rather than Select lets us hold less data in memory during the Raw-to-embedded conversion process. That memory performance is extremely unlikely to matter. In theory we can use Select to scan a slice, but unless we can find a solution for scanning embedded structs we have to do a loop to convert types either way. There is something to be said for code consistency in using Select, on the other hand this is what Queryx is for. I can see arguments for either usage. My preference leans a bit to Queryx and since the value proposition is low to invest in a Select query (without a solution for embedded struct scanning) I left it as-is pending significantly strong feelings from the team.

Reply:
> unless we can find a solution for scanning embedded structs we have to do a loop to convert types either way

I see what you mean! I think `sqlx.Select` would only really simplify the `rows.StructScan(&rawResult)` call (which is, under the hood, what those sqlx struct methods call as well)

I think we should probably have a sync today to brainstorm some alternative approaches to this to see if there's anything we can do to ease the amount of wiring we have to do to get these results turned into concrete types.

(Part of me is tempted to create a new type to represent this joined data and use that instead of _actual_ ModelPlan and OperationalSolution types)